FHEM/00_SIGNALduino.pm

```diff
@@ -2534,11 +2534,14 @@ SIGNALduino_Parse_MC($$$$@)
 			Debug "$name: extracted data $bitData (bin)\n" if ($debug); ## Convert Message from hex to bits
 		   	SIGNALduino_Log3 $name, 5, "$name: extracted data $bitData (bin)";
 		   	
-		   	my $method = $ProtocolListSIGNALduino{$id}{method};
-		    if (!exists &$method)
-			{
-				SIGNALduino_Log3 $name, 5, "$name: Error: Unknown function=$method. Please define it in file $0";
+	   	my $method;
+	    if (!exists $ProtocolListSIGNALduino{$id}{method}) {
```

Comment:
Die Intention verstehe ich, allerdings würde ich es etwas anders lösen.
Ich glaube wir hatten einen vergleichbaren Fall auch schon mal. Ich hoffe ich finde die codestelle.

Die Defaults würde ich 1x beim Einlesen der Daten setzen und nicht bei jeder Verarbeitung.

Reply:
> Die Defaults würde ich 1x beim Einlesen der Daten setzen und nicht bei jeder Verarbeitung.

korrigierst du den code wenn du die Stelle fandest?

Reply:
Ja, ich hoffe es eilt nicht zu sehr :)

Reply:
Die Stelle habe ich Gefunden.... allerdings ist das noch in 00_SIGNALduino.pm

Das passt da aber nicht gut hin...

Ihh bin jetzt etwas hin und her gerissen.

1) Die Protokolldaten werden über lib::SD_Protocols verwaltet.

Ich könnte dort eine Funktion einbauen die nach LoadHash aufgerufen wird.
In etwa sowas wie `setDefaut`. Dann werden die Standardwerte im Hash hinterlegt.
Das Abrufen der Daten würde ich auf `getProperty($id,"method"); `ändern.

+ Die Werte müssen nur einmal gesetzt werden und sind dann immer vorhanden
- 00_SIGNALduino.pm hat keine Kontrolle mehr über die Daten.

2) In 00_SIGNALduino.pm wird der Wert von Method via `$method= checkProperty($id,"method",\&main::SIGNALduino_MCRAW);`
Gesetzt. Dann wird zur Laufzeit verifiziert ob es den method Wert gibt und dann dieser oder der default zurückgegeben.

+ Die Zugriffe auf die Protokolldaten würde ich ohnehin umstellen wollen
- Das prüfen eines Inhaltes zur Laufzeit kostet ein paar CPU Zyklen.

Reply:
Was du vorhast, habe ich verstanden. Was durchaus besser oder CPU sparender ist, kann ich schwer einschätzen.

Das ein Standard gesetzt wird, wenn Eintrag noch vorhanden ist, finde ich gut.

Reply:
Ich denke, den Standard kann man beim Laden der Protokolldaten setzen. Das Programm funktioniert ja ohne die Methode nicht.

Deinen PR müsste ich dafür aber ein bisschen erweitern :)